### ISSUE TITLE
Automatic support for a conda or venv-based activate -> work -> deactivate workflow in coc-pyright

### ISSUE THREAD

**Issue by anjos** [fannheyward/coc-pyright#662]
I'm following the advice from the README in this package.  I'm however working with conda environments instead of virtualenv.

TL;DR: This package could have a flag like `"python.autoDetectVenv": true` that tries its best to determine the path to the Python interpreter as well as the location of python modules.  Activate scripts from virtuenv provide `$VIRTUAL_ENV`.  Activate scripts from conda provide `$CONDA_PREFIX`.  This could automate the setup in most cases.

I have a "python" program I set on my own `coc-settings.json` with `"python.pythonPath": "<MY PATH>",`, I activate the environment and start editing a python file.  I checked with a log file this mechanism works and I can correctly see everything is detected as expected (the right python interpreter, from the conda venv is launched).  However, this does not work as expected, and pyright still cannot find the modules in the Python (conda) environment. (note. `CocCommand pyright.version` shows `[coc.nvim] coc-pyright 1.1.185 with Pyright 1.1.188`.  I'm using neovim, version 0.5.1)

I tried another path: I set `"python.venvPath": "<PATH TO CONDA ENVS>"` in my own central `coc-settings.json`, and then, on the root of my project, I edit `pyproject.toml` to contain a `[tool.pyright]` section with the right name of the environment (as described here: https://github.com/microsoft/pyright/blob/main/docs/configuration.md).  This works better, so the editor starts resolving the symbols correctly.

Unfortunately, the project I work on is shared with other developers.  If I edit `pyproject.toml` to contain the name of my personal venv , this may not be same across all developers and this will create issues.

Conda exports a number of environment variables when it activates.  Wouldn't it be possible to have this plugin automatically scan for these known environment variables and setup the correct parameters accordingly?

**Comment by fannheyward** [comment]
> Conda exports a number of environment variables when it activates

I didn't use Conda, can you share me these env variables?

**Comment by anjos** [comment]
There is a bunch of those, with specific needs.  I think the most useful are the following:

1. `$CONDA_PREFIX`:  That is the equivalent of `$VIRTUAL_ENV` in pure-python virtual environments, it is the prefix to the activated environment.  `$CONDA_PREFIX/lib/python*/site-packages` contains the modules.
2. `$CONDA_DEFAULT_ENV`: The currently activated environment (this would be equivalent of the "venv" setting in pyright)

If you want to have a precise calculation of the module locations you can do something like `$CONDA_PREFIX/bin/python -c 'print(__import__("site").getsitepackages()[0])'`.  Does that help?

**Comment by fannheyward** [comment]
I've made an improve in v1.1.189, please give a try.

**Comment by anjos** [comment]
It works flawlessly.  Thanks!  It is now possible to follow the workflow without any previous configuration, except for activating the environment.
